lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -3,24 +3,58 @@ module Compute
     class Ovirt
       class V4
         class Real
+          DEFAULT_PROVISIONED_SIZE = 1 * 2**30
+          DISK_INTERFACES = {
+            "virtio" => OvirtSDK4::DiskInterface::VIRTIO,
+            "virtio_scsi" => OvirtSDK4::DiskInterface::VIRTIO_SCSI,
+            "virtio_vscsi" => OvirtSDK4::DiskInterface::VIRTIO_VSCSI,
+            "ide" => OvirtSDK4::DiskInterface::IDE
+          }.freeze
+
           def check_for_option(opts, name)
             opts[name.to_sym] || opts[(name + "_name").to_sym]
           end
 
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def process_vm_opts(opts)
-            return unless check_for_option(opts, "template") && check_for_option(opts, "storagedomain")
+            return unless check_for_option(opts, "template") && check_for_option(opts, :disks)
 
             template_id = opts[:template] || client.system_service.templates_service.search(:name => opts[:template_name]).first.id
-            template_disks = client.system_service.templates_service.template_service(template_id).get.disk_attachments
-            storagedomain_id = opts[:storagedomain] || storagedomains.select { |s| s.name == opts[:storagedomain_name] }.first.id
+            template_disks = client.system_service.templates_service.template_service(template_id).disk_attachments_service.list
 
             # Make sure the 'clone' option is set if any of the disks defined by
             # the template is stored on a different storage domain than requested
```

Comment:
This comment explains why there used to be use of `storagedomain_id`, it should be removed if you are removing the usecase of changed storage domain.
Also, for the record - we now detect any change including storage domain and add the `clone` option in: https://github.com/theforeman/foreman/pull/6110 which is why this code can be removed.